client/src/telemetry/glean-context.tsx

```diff
@@ -0,0 +1,116 @@
+import * as React from "react";
+import { path, pageEvent, referrer } from "./generated/page";
+import { clicked } from "./generated/event";
+import * as pings from "./generated/pings";
+import * as Glean from "@mozilla/glean/web";
+
+export type PageProps = {
+  referrer: string | undefined;
+  path: string | undefined;
+};
+
+export type PageEventProps = {
+  referrer: string | undefined;
+  path: string | undefined;
+};
+
+export type EventProps = {
+  label: string;
+  position: string;
+  type: string;
+};
+
+export type GleanAnalytics = {
+  page: (arg: PageProps) => void;
+  pageEvent: (arg: PageEventProps) => void;
+  event: (arg: EventProps) => void;
+};
+
+const FIRST_PARTY_DATA_OPT_OUT_COOKIE_NAME = "moz-1st-party-data-opt-out";
+const GLEAN_APP_ID = "mdn-yari";
+
+function glean(): GleanAnalytics {
+  if (typeof window === "undefined") {
+    //SSR return noop.
+    return {
+      page: (page: PageProps) => {},
+      pageEvent: (page: PageEventProps) => {},
+      event: (event: EventProps) => {},
+    };
+  }
+  const channel = process.env.REACT_APP_GLEAN_CHANNEL || "stage";
+  const debugPings = Boolean(
+    JSON.parse(process.env.REACT_APP_GLEAN_DEBUG || "null")
+  );
+  const gleanEnabled = !Boolean(
+    JSON.parse(process.env.REACT_APP_GLEAN_DISABLED || "null")
+  );
+
+  let userIsOptedOut = false;
+
+  const value = `; ${document.cookie}`;
+
+  const parts = value.split(`; ${FIRST_PARTY_DATA_OPT_OUT_COOKIE_NAME}=`);
+  if (parts.length === 2) {
+    userIsOptedOut = !!parts.pop()?.split(";").shift();
+  }
```

Comment:
this is rather esoteric, could we do something like:
```js
const cookies = `; ${document.cookie};`;
const userIsOptedOut = cookies.includes(`; ${FIRST_PARTY_DATA_OPT_OUT_COOKIE_NAME}=1;`);
```

Reply:
Absolutely.